src/lib/KeyShares/KeyShares.ts

```diff
@@ -51,54 +31,63 @@ export class KeyShares {
   @ValidateNested()
   public payload: KeySharesPayload;
 
+  constructor() {
+    this.data = new KeySharesData();
+    this.payload = new KeySharesPayload();
+  }
+
   /**
-   * @param version
+   * Build payload from encrypted shares, validator public key and operator IDs
+   * @param publicKey
+   * @param operatorIds
+   * @param encryptedShares
    */
-  constructor({ version }: { version: string }) {
-    this.version = version;
-    this.data = this.getByVersion('data', version);
-    this.payload = this.getByVersion('payload', version);
+  buildPayload(metaData: IPayloadMetaData): any {
+    return this.payload.build({
+      publicKey: metaData.publicKey,
+      operatorIds: operatorSortedList(metaData.operators).map(operator => operator.id),
+      encryptedShares: metaData.encryptedShares,
+    });
   }
 
   /**
-   * Set final payload for web3 transaction and validate it.
-   * @param payload
+   * Build shares from bytes string and operators list length
+   * @param bytes
+   * @param operatorCount
    */
-  generateContractPayload(data: any): KeySharesPayload {
-    const payloadData = this.payload.build(data);
-    this.payload?.setData(payloadData);
-
-    return this.payload;
+  buildSharesFromBytes(bytes: string, operatorCount: number): any {
+    bytes = bytes.replace('0x', '');
+    const pkLength = parseInt(bytes.substring(0, 4), 16);
+
+    // get the public keys part
+    const pkSplit = bytes.substring(4, pkLength + 2);
+    const pkArray = ethers.utils.arrayify('0x' + pkSplit);
+    const sharesPublicKeys = this._splitArray(operatorCount, pkArray).map(item =>
+      ethers.utils.hexlify(item),
+    );
+
+    const eSplit = bytes.substring(pkLength + 2);
+    const eArray = ethers.utils.arrayify('0x' + eSplit);
+    const encryptedKeys = this._splitArray(operatorCount, eArray).map(item =>
+      Buffer.from(ethers.utils.hexlify(item).replace('0x', ''), 'hex').toString(
+        'base64',
+      ),
+    );
+    return {
+      sharesPublicKeys,
+      encryptedKeys,
+    };
```

Comment:
We remember to not change the way we join/split the shares including length prefix etc, right? Otherwise it will break entire flow with web and ssv nodes. Just double checking it.